api/v1/installation_types.go

```diff
@@ -198,8 +198,30 @@ type InstallationSpec struct {
 	// Kubernetes Service CIDRs. Specifying this is required when using Calico for Windows.
 	// +optional
 	ServiceCIDRs []string `json:"serviceCIDRs,omitempty"`
+
+	// Azure is used to configure azure provider specific options.
+	// +optional
+	Azure *Azure `json:"azure,omitempty"`
+}
+
+type Azure struct {
+	// PolicyMode determines whether the "control-plane" label is applied to namespaces. It offers two options: Default and Manual.
+	// The Default option adds the "control-plane" label to namespaces that have the PodSecurityStandard set to privileged.
+	// The Manual option does not apply the "control-plane" label to any namespace.
+	// Default: Default
+	// +optional
+	// +kubebuilder:validation:Enum=Default;Manual
+	// +kubebuilder:default:=Default
+	PolicyMode *PolicyMode `json:"policyMode,omitempty"`
```

Comment:
Just wondering why this field is a pointer exactly. The validation and the controller will fill this field if the struct isn't nil. I think it is never possible to be nil in practice.

Reply:
Any optional field in the API should be a pointer - it is possible to send a yaml like this:

```
spec:
  azure: {}
```

In which case it will be `nil` in the API server and defaulted using the kubebuilder tag.

This is important for the forward looking chance that we add other fields to this sub-struct as well, where someone may want to set another field in the Azure section and our code expects this to be set.

Reply:
> may want to set another field in the Azure section and our code expects this to be set.

If we introduce field azure.foo and you apply `azure: {foo: bar}`, kube builder, the policyMode field will be set once our controllers are retrieving the CR.

Other than that I guess my question is answered; we should keep the pointer.